examples/simulation/simulate_telescope.cpp

```diff
@@ -0,0 +1,128 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/edm/track_parameters.hpp"
+#include "traccc/io/utils.hpp"
+#include "traccc/options/common_options.hpp"
+#include "traccc/options/handle_argument_errors.hpp"
+
+// detray include(s).
+#include "detray/detectors/create_telescope_detector.hpp"
+#include "detray/simulation/simulator.hpp"
+#include "detray/simulation/track_generators.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// System include(s).
+#include <filesystem>
```

Comment:
:confused: Wouldn't you need to link against `-lstdc++fs` to make this work with GCC 9? Is our CI not building this code? Since I thought we used GCC 9 in most of our CI builds. :confused:

I'm generally not the biggest fan of [std::filesystem](https://en.cppreference.com/w/cpp/filesystem) exactly because it used to be very finicky to use in practice. I believe with the latest versions of GCC and Clang it's a lot easier to use. But I still wonder if we should use it in this project over [boost::filesystem](https://www.boost.org/doc/libs/master/libs/filesystem/doc/index.htm). Since the latter is a lot more reliable vis-a-vis compiler versions.

Reply:
> Is our CI not building this code? 

It is not a part of CI. Maybe that's why the CI passed. I didn't know that `std::filesystem` has that kind of issue. I changed to use boost filesystem.